lib/mozart_fetcher/fetcher.ex

```diff
@@ -11,20 +11,35 @@ defmodule MozartFetcher.Fetcher do
 
   def process(components) do
     ExMetrics.timeframe "function.timing.fetcher.process" do
-      max_timeout =
-        TimeoutParser.max(components) +
-          Application.get_env(:mozart_fetcher, :additional_task_await_timeout)
+      max_timeout = TimeoutParser.max(components)
+      stream_opts = [timeout: max_timeout, on_timeout: :kill_task, max_concurrency: 40]
 
       components
       |> Enum.with_index()
-      |> Enum.map(&Task.async(fn -> Component.fetch(&1) end))
-      |> Enum.map(&Task.await(&1, max_timeout))
-      |> decorate_response
+      |> Task.async_stream(&Component.fetch/1, stream_opts)
+      |> Enum.to_list()
+      |> extract_successful()
+      |> decorate_response()
       |> Jason.encode!()
     end
   end
 
   defp decorate_response(envelopes) do
     %{components: envelopes}
   end
+
+  # returns the successful one (including 500 etc) and logs
+  # the tasks which have not completed.
```

Comment:
Can we return an empty envelope If they haven't completed, a bit like requester does, or just map the id to a 408 or 500 status as we don't have any content.

I was also thinking about the ordering, that we return an index for the component, but if we are missing some, I assume they will just not be on the page. I was wondering if there were any other issues with that as it is different behaviour from previously.